libantworld/renderable.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+//----------------------------------------------------------------------------
+// BoBRobotics::AntWorld::Renderable
+//----------------------------------------------------------------------------
+namespace BoBRobotics
+{
+namespace AntWorld
+{
+class Renderable
+{
+public:
+    virtual ~Renderable(){}
+
+    //----------------------------------------------------------------------------
+    // Declared virtuals
+    //----------------------------------------------------------------------------
+    virtual void render() const = 0;
```

Comment:
It seems a bit weird that this is ``const``. Surely the internal state of ``Renderable``s changes when you render them?

Reply:
I'm not 100% sure about this bit of design at all tbh but render really shouldn't be changing any state - it should just making some opengl draw calls.